docs/design/lambdas.md

```diff
@@ -0,0 +1,481 @@
+# Lambdas
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Syntax Overview](#syntax-overview)
+    -   [Return type](#return-type)
+        -   [Return expression](#return-expression)
+        -   [Explicit return type](#explicit-return-type)
+        -   [No return](#no-return)
+    -   [Implicit parameters in square brackets](#implicit-parameters-in-square-brackets)
+    -   [Parameters](#parameters)
+    -   [Syntax defined](#syntax-defined)
+-   [Positional parameters](#positional-parameters)
+    -   [Positional parameter restrictions](#positional-parameter-restrictions)
+-   [Captures](#captures)
+    -   [Capture modes](#capture-modes)
+    -   [Default capture mode](#default-capture-mode)
+-   [Function fields in lambdas](#function-fields-in-lambdas)
+-   [Copy semantics](#copy-semantics)
+-   [Self and recursion](#self-and-recursion)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Syntax Overview
+
+One goal of Carbon's lambda syntax is to have continuity between lambdas and
+function declarations. Below are some example declarations:
+
+Implicit return types:
+
+```carbon
+// In a variable:
+let lambda: auto = fn => T.Make();
```

Comment:
This document uses `fn =>` here and `fn ->` below, I assume one of those isn't correct?

Reply:
Oh, never mind! I see this gets explained.